### DISCUSSION TITLE
User session is not created for the grant type 'refresh_token' (refresh_token_type=Offline), why?

### DISCUSSION THREAD

**Discussion by prakashpkt20** [keycloak/keycloak#26606] [Q&A] [answered]
Whenever we create offline token in keycloak an userSession will be created, and the session is visible in 'Sessions' tab of the user (Assume the 'Offline session Idle' is 30 days and obviously userSession will not be retained for 30 days & it will be wiped off after some number of days.). So when we try token exchange(grant_type as 'urn:ietf:params:oauth:token-type:access_token') or grant type with 'refresh_type'  to get access token, new userSession is not created.

**Comment by sschu** [comment, marked as answer]
Because internally this is an offline user session that is no longer bound to a normal user session once the normal session expires.
